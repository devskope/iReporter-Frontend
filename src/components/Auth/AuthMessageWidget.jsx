/* eslint-disable react/no-array-index-key */
import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const AuthMessageWidget = ({ errors, clearErrors }) => {
  const className = classNames({
    auth__messages: true,
    'auth__messages--error visible': errors.length > 0,
  });

  const messages = errors.map((message, i) => (
    <li className="auth__message" key={i}>
      {message}
    </li>
  ));

  return (
    <div className={className}>
      <button
        className="auth__messages-dismiss"
        onClick={clearErrors}
        type="button"
      >
        ‚äù
      </button>
      <ul className="auth__message-list">{messages}</ul>
    </div>
  );
};

AuthMessageWidget.propTypes = {
  clearErrors: PropTypes.func.isRequired,
  errors: PropTypes.arrayOf(PropTypes.string).isRequired,
};

export default AuthMessageWidget;
