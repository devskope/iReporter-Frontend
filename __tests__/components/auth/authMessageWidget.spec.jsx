import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import AuthMessageWidget from '../../../src/components/Auth/AuthMessageWidget';

const setup = () => {
  const props = {
    clearErrors: jest.fn(),
    errors: ['error message', 'error message'],
  };
  return shallow(<AuthMessageWidget {...props} />);
};

describe('AuthMessageWidget', () => {
  it('should render without crashing', () => {
    const wrapper = setup();

    expect(wrapper.find('.auth__message').exists()).toBe(true);
    expect(wrapper.find('.auth__message-list').exists()).toBe(true);
  });

  it('should match snapshot', () => {
    const wrapper = setup();

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
