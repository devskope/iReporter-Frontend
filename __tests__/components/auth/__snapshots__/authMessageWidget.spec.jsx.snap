// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthMessageWidget should match snapshot 1`] = `ShallowWrapper {}`;
